src/database/queries.rs

```diff
@@ -59,19 +59,24 @@ pub(crate) async fn missing_blocks(conn: &mut PgConnection) -> Result<Vec<u32>>
     .collect())
 }
 
+/// Find missing blocks from the relational database between numbers `min` and
+/// MAX(block_num). LIMIT result to length `max_block_load`.
 pub(crate) async fn missing_blocks_min_max(
     conn: &mut PgConnection,
     min: u32,
+    max_block_load: u32,
 ) -> Result<HashSet<u32>> {
     Ok(sqlx::query_as::<_, (i32,)>(
         "SELECT generate_series
         FROM (SELECT $1 as a, max(block_num) as z FROM blocks) x, generate_series(a, z)
         WHERE
         NOT EXISTS(SELECT id FROM blocks WHERE block_num = generate_series)
         ORDER BY generate_series ASC
+        LIMIT $2
         ",
     )
     .bind(min as i32)
+    .bind(max_block_load as i32)
```

Comment:
Not a fan of these casts here. If `max_block_load` is bigger than `i32::MAX` [it will wrap around to a negative number](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=057ba7218ff448ee8845be9973cd021c).
Is this something `sqlx` requires us to do or can we use `u32` throughought? If we can't then we should use `try_into` here.

Reply:
will require `try_into`, since there's no direct PostgreSQL equivalent type to an unsigned integer sqlx chooses not to represent it when binding values.

Reply:
Can we just cast to an i64 (bigint in postgres)?

Reply:
I believe casting to i64 would work too